src/app.jsx

```diff
@@ -73,9 +75,11 @@ export const Application = () => {
             });
 
             setFiles([]);
+            setDirItemsCnt(0);
             channel.current.addEventListener("message", (ev, data) => {
                 const item = JSON.parse(data);
                 if (item.event === "present") {
+                    setDirItemsCnt(c => c !== undefined ? c + 1 : 0);
```

Comment:
Same issue -- don't call setDirItemsCnt() n times in this handler, but only once in the end in "ready".